### PR TITLE
EUI-7430

### PR DESCRIPTION
**Pull request by MateiHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-7430

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Review by DavidJayakumar — DISMISSED** [review]

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
baseLocation should not be being changed to primaryLocation as is shown within these changes  - needs to be kept as baseLocation

**Review by connorpgpmcelroy — DISMISSED** [review]
LGTM

**Review by udaydenduluri33 — APPROVED** [review]
LGTM

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Review by udaydenduluri33 — APPROVED** [review]

File: api/user/utils.ts [hunk 1]
```diff
@@ -29,11 +29,10 @@ export function isCurrentUserCaseAllocator(currentUserRoleAssignment: RoleAssign
...
-   && currentUserRoleAssignment.roleType === ORGANISATION_ROLE_TYPE
-   && currentUserRoleAssignment.roleName === CASE_ALLOCATOR_ROLE
-   && (!currentJurisdiction || currentUserRoleAssignment.attributes.jurisdiction === currentJurisdiction)
-   // primaryLocation changed to baseLocation here
-   && (!currentLocation || currentUserRoleAssignment.attributes.baseLocation === currentLocation);
+    && currentUserRoleAssignment.roleType === ORGANISATION_ROLE_TYPE
+    && currentUserRoleAssignment.roleName === CASE_ALLOCATOR_ROLE
+    && (!currentJurisdiction || currentUserRoleAssignment.attributes.jurisdiction === currentJurisdiction)
+    && (!currentLocation || currentUserRoleAssignment.attributes.baseLocation === currentLocation);
 }
```
Comment (phillip-whitaker-hmcts) [inline api/user/utils.ts:36]: it seems in this method nothing has been changed apart from removing a comment, I would try and refrain from doing this to simplify the code reviews.

File: api/workAllocation/util.ts [hunk 2]
```diff
@@ -354,20 +354,24 @@ export async function getCaseIdListFromRoles(roleAssignmentList: RoleAssignment[
...
 
+export function filterMyAccessRoleAssignments(roleAssignmentList: RoleAssignment[]) {
+  return roleAssignmentList.filter(roleAssignment =>
+    (
+      roleAssignment.grantType === 'SPECIFIC' ||
+      roleAssignment.roleName === 'specific-access-requested' ||
+      roleAssignment.roleName === 'specific-access-denied'
+    ) &&
+    (!roleAssignment.attributes || roleAssignment.attributes.substantive !== 'Y')
+  );
```
Comment (phillip-whitaker-hmcts) [inline api/workAllocation/util.ts:369]: The predicate in this filter method is complex. Try and break it down. There really should be a maximum of 2-3 equality checks within a single statement for readability.

**Pull request merged**